FLAnimatedImage/BlendedImageCache.m

```diff
@@ -0,0 +1,90 @@
+//
+//  BlendedImageCache.m
+//  FLAnimatedImage
+//
+//  Created by Sih Ou-Yang on 2024/2/23.
+//  Copyright © 2024 com.cardinalblue. All rights reserved.
+//
+
+#import <Foundation/Foundation.h>
+#import <UIKit/UIKit.h>
+
+@interface BlendedImageInfo : NSObject
+
+@property(nonatomic) UIImage *image;
+@property(nonatomic) NSTimeInterval timestamp;
+
+@end
+
+@implementation BlendedImageInfo
+
+@end
+
+@interface BlendedImageCache : NSObject
+
+- (instancetype)initWithLimit:(NSInteger)limit;
+
+- (void)add:(UIImage *)image withIndex:(NSInteger)index;
+- (nullable UIImage *)imageAtIndex:(NSInteger)index;
+
+
+@end
+
+@interface BlendedImageCache ()
+
+@property(nonatomic) NSMutableDictionary<NSNumber *, BlendedImageInfo *> *blendedImageDict;
+
+// limit: 0 represents no cache
+@property(nonatomic) NSInteger limit;
+
+@end
+
+@implementation BlendedImageCache
+
+-(instancetype)initWithLimit:(NSInteger)limit {
+    self = [super init];
+    if (self) {
+        _blendedImageDict = [[NSMutableDictionary alloc] init];
+        _limit = limit;
+    }
+    return self;
+}
+
+- (void)add:(UIImage *)image withIndex:(NSInteger)index {
+    if (self.limit == 0) {
+        return;
+    }
+    // if cache reaches limit -> remove the oldest
+    if (self.blendedImageDict.count == self.limit) {
+        [self removeOldestCache];
+    }
+
+    // add to cache
+    BlendedImageInfo *info = [[BlendedImageInfo alloc] init];
+    info.image = image;
+    info.timestamp = [NSDate new].timeIntervalSince1970;
+    self.blendedImageDict[@(index)] = info;
+}
+
+- (UIImage *)imageAtIndex:(NSInteger)index {
+    return self.blendedImageDict[@(index)].image;
```

Comment:
Perhaps we could consider updating the timestamp upon access, ensuring that the removal logic is based on the oldest cache entry that hasn't been used.

Reply:
Updated the functionality to the caller of this method.